lib/needs_exporter.rb

```diff
@@ -10,23 +10,25 @@ def initialize
   end
 
   def run
+    count = @needs.count
     @needs.order_by(:_id.asc).each_with_index do |need, index|
       export(need, index, count)
     end
   end
 
 private
 
-  def export(need, index)
+  def export(need, index, count)
     slug = generate_slug(need)
     need_revision_groups = need_revision_groups(need.revisions)
     snapshots = need_revision_groups.map { |nrg| present_need_revision_group(nrg, slug)}
     compute_superseded_needs(snapshots)
     @api_client.import(need.content_id, snapshots)
     links = present_links(need)
     @api_client.patch_links(need.content_id, links: links)
-    p "#{index}/#{@needs.count}"
-    p "exported #{slug}"
+
+    padding = Math.log10(count).ceil
```

Comment:
Its used in the string on the line below, maybe width would be a better name, as that is what its called in the format string documentation [1].

`%#{padding}d` will become something like `%4d` which will be substituted by format with the value given (`index + 1`), which will then become something like `__14` (where the underscores are actually spaces), or `3000` with padding on the left of the string such that its 4 characters in width.

The desired width (or padding as its called above) is the number of characters required to display the largest possible index. The largest possible index is equal to the count, and to get the number of digits required to display it in base 10, you take the logarithm of that value in base 10, and round up.

All of this just gives you output formatted such that it moves around less, making it slightly easier to read. 

1: https://ruby-doc.org/core-2.2.0/Kernel.html#method-i-format